test/jdk/java/lang/instrument/VerifyLocalVariableTableOnRetransformTest.java

```diff
@@ -120,8 +137,92 @@ public VerifyLocalVariableTableOnRetransformTest(String name)
         // Table (LVT) attribute so the class file bytes seen by the
         // retransformClasses() call will not match the class file bytes
         // seen at initial class load time.
-        assertTrue(fTargetClassName + " did not match .class file",
-            fTargetClassMatches);
+        compareClassFileBytes(false);
+    }
+
+    private File originalTargetClassFile() {
+        String resourceName = fTargetClassName + ".class";
+        return new File(System.getProperty("test.classes", "."), resourceName);
+    }
+
+    private File transformClassFile() {
+        // This file will get created in the test execution
+        // directory so there is no conflict with the file
+        // in the test classes directory.
+        String resourceName = fTargetClassName + ".class";
+        return new File(resourceName);
+    }
+
+    private static final String[] expectedDifferentStrings = {
+            "^Classfile .+$",
+            "^[\\s]+SHA-256 checksum .[^\\s]+$"
+    };
+
+    private boolean expectedDiffenent(String line) {
+        for (String s: expectedDifferentStrings) {
+            Pattern p = Pattern.compile(s);
+            if (p.matcher(line).find()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void compareClassFileBytes(boolean initialLoad) throws Throwable {
+        if (fTargetClassMatches) {
+            return;
+        }
+        File f1 = originalTargetClassFile();
+        File f2 = transformClassFile();
+        System.out.println(fTargetClassName + " did not match .class file");
+        System.out.println("Disassembly difference (" + f1 + " vs " + f2 +"):");
+        // compare 'javap -v' output for the class files
+        List<String> out1 = disassembleClassFile(f1);
+        List<String> out2 = disassembleClassFile(f2);
+        boolean different = false;
+        boolean orderChanged = false;
+        int lineNum = 0;
+        for (String line: out1) {
+            if (expectedDiffenent(line)) {
```

Comment:
Is it possible to use the `expectedDiffenent()` as a filter to get only needed lines from `disassembleClassFile()`?

Reply:
It's implemented this way to keep lineNum correct. But it was error here. Fixed now

Reply:
And fixed typo in the method name